src/pages/home/report/ReportActionItem.tsx

```diff
@@ -73,198 +76,219 @@ import ReportActionItemMessage from './ReportActionItemMessage';
+            if (ReportActionContextMenu.isActiveReportAction(action.reportActionID)) {
                 ReportActionContextMenu.hideContextMenu();
                 ReportActionContextMenu.hideDeleteModal();
             }
-            if (EmojiPickerAction.isActive(props.action.reportActionID)) {
+            if (EmojiPickerAction.isActive(action.reportActionID)) {
                 EmojiPickerAction.hideEmojiPicker(true);
             }
-            if (reactionListRef.current && reactionListRef.current.isActiveReportAction(props.action.reportActionID)) {
-                reactionListRef.current.hideReactionList();
+            if (reactionListRef?.current?.isActiveReportAction(action.reportActionID)) {
+                reactionListRef?.current?.hideReactionList();
             }
         },
-        [props.action.reportActionID, reactionListRef],
+        [action.reportActionID, reactionListRef],
     );
 
     useEffect(() => {
         // We need to hide EmojiPicker when this is a deleted parent action
-        if (!isDeletedParentAction || !EmojiPickerAction.isActive(props.action.reportActionID)) {
+        if (!isDeletedParentAction || !EmojiPickerAction.isActive(action.reportActionID)) {
             return;
         }
 
         EmojiPickerAction.hideEmojiPicker(true);
-    }, [isDeletedParentAction, props.action.reportActionID]);
+    }, [isDeletedParentAction, action.reportActionID]);
 
     useEffect(() => {
-        if (!_.isUndefined(prevDraftMessage) || _.isUndefined(props.draftMessage)) {
+        if (!!prevDraftMessage || !draftMessage) {
             return;
         }
 
         focusTextInputAfterAnimation(textInputRef.current, 100);
-    }, [prevDraftMessage, props.draftMessage]);
+    }, [prevDraftMessage, draftMessage]);
 
     useEffect(() => {
         if (!Permissions.canUseLinkPreviews()) {
             return;
         }
 
-        const urls = ReportActionsUtils.extractLinksFromMessageHtml(props.action);
-        if (_.isEqual(downloadedPreviews.current, urls) || props.action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+        const urls = ReportActionsUtils.extractLinksFromMessageHtml(action);
+        if (
+            (downloadedPreviews.current.length === urls.length && downloadedPreviews.current.every((value, arrIndex) => value === urls[arrIndex])) ||
+            action.pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE
+        ) {
             return;
         }
 
         downloadedPreviews.current = urls;
-        Report.expandURLPreview(props.report.reportID, props.action.reportActionID);
-    }, [props.action, props.report.reportID]);
+        Report.expandURLPreview(report.reportID, action.reportActionID);
+    }, [action, report.reportID]);
 
     useEffect(() => {
-        if (_.isUndefined(props.draftMessage) || !ReportActionsUtils.isDeletedAction(props.action)) {
+        if (!draftMessage || !ReportActionsUtils.isDeletedAction(action)) {
```

Comment:
```suggestion
        if (draftMessage === undefined || !ReportActionsUtils.isDeletedAction(action)) {
```

For safety, condition will pass if `draftMessage` is `''` which wouldn't happen in the old code